proxy/config.go

```diff
@@ -53,7 +53,10 @@ type Config struct {
 	// Encryption configuration
 	// --
 
-	TLSConfig            *tls.Config    // necessary for TLS, HTTPS, QUIC
+	TLSConfig *tls.Config // necessary for TLS, HTTPS, QUIC
+
+	DoHClientTLSConfig *tls.Config // necessary for DoH Client Authentication
```

Comment:
Do we really need to limit this to DOH?

DoQ and DoT may also support client auth, let's rename it to `TLSClientConfig` and change the code accordingly so that DoQ/DoT upstreams supported it.